app/commands/v2/patch_link_set.rb

```diff
@@ -18,17 +18,25 @@ def call
         lock_version.save!
 
         grouped_links.each do |group, payload_content_ids|
+          # For each set of links in a LinkSet scoped by link_type, this iterator
+          # deletes the entire existing set and then imports all the links in the
+          # payload, preserving their ordering. The activerecord-import gem is
+          # used to create a single SQL statement for importing all links.
           links = link_set.links.where(link_type: group)
           existing_content_ids = links.pluck(:target_content_id)
 
-          content_ids_to_create = payload_content_ids - existing_content_ids
-          content_ids_to_delete = existing_content_ids - payload_content_ids
+          links.where(target_content_id: existing_content_ids).delete_all
```

Comment:
Yup, that's one downside to this approach. I checked this with @tijmenb and the view is this shouldn't be a problem since we don't rely on it for anything on our side. Not sure if you use that field elsewhere.